assets/base.css

```diff
@@ -2741,3 +2741,7 @@ details-disclosure > details {
   border-left: none;
   border-right: none;
 }
+
+.isolate {
+  isolation: isolate;
+}
```

Comment:
So I guess what I didn't realize was how many places this fix (isolate or zindex/position) was needed now with the color scheme changes on most sections. This isolation fix is a great replacement for what we're currently doing but I don't think either strategy is going to cut it in the long run. Reseting the z context in anyway ruins the relationship between peer elements. So this technique we're using fixes elements _within_ the new context, but breaks the relationship between sections.
<img src="https://screenshot.click/19-12-riqep-01v15.png" width="400"/>

One area where I know we need this to work properly is image with text, which fortunately doesn't need this fix or have this problem, but with the ability to reduce section spacing on any section I think we need to fix this another way (right?). Really I think our problem is with the background color. We may need to look at relocating this maybe also be on it's own pseudo element with a lower z-index or something 😅 

Does this make sense? Am I missing something
cc: @tyleralsbury @ludoboludo

Reply:
> Maybe in this case we can just keep it for the buttons/inputs
This might be safe, but yeah I think we'll need to come up with something else for any more significant elements.

Reply:
I can revert it for `.card .card__inner .card__media` and keep it everywhere else

Reply:
Well so it's not just cards. If you put the isolate class (or z-index/position) on a multicolumn parent, it's changing the context there too for the content container shadows. I believe (if im not confusing myself) if we fix the section background color dilemma, we won't need any of z context changes like this on buttons or otherwise.

Reply:
In that case, we would only apply `isolate` to product recommendations, featured product, contact form, localization selectors and newsletter since theyre the only ones without cards

Reply:
> Or perhaps we can add isolate to a higher level to create a new stacking context between the sections and avoid what you have in the screenshot.

Feel free to try it, I definitely could be missing something obvious here, but I think the problem is we want shadows to be below everything anywhere on the page except background color, but z-index: -1 puts it behind _everything_ including the elements with a background color. So those 2 things are at odds.

I'm curious what Tyler or Ludo thinks, but I guess I'm not against merging this as is just because it shouldn't be breaking anything that isn't already "broken" on main.

Reply:
So that fix was originally needed and added because of when we were using a gradient as the background. When it's set to background color it was fine but when using a gradient, the shadow of the elements within the section would disappear. 
If `isolate` is not changing the current behaviour, it's maybe a good thing to update to reduce the amount of CSS but doesn't feel like a high priority as it's a refactor vs a fix. 

About the relationship between section and specifically when the shadow would potentially overlap it. Were we maybe thinking of adding some spacing dynamically based on the vertical offset value for the shadow ? Cause if we did then it that would kinda solve our issue no ? So if the value is negative then apply some spacing to the top and to the bottom when positive. Tho it can't be perfect either due to the `blur` effect/value 🤔

Reply:
They can use a combination of vertical spacing between sections and padding settings to adjust things.  I don't know if it's possible for us to account for all of the most extreme circumstances in a way that doesn't create wacky spaghetti code all over the place.

We do need to weigh technical debt against perfection, and if there are already settings that can alleviate the majority of expected cases I think we're okay with some of the more out there things not being 100% perfect.  We can create an issue to explore it, but it's the sort of thing that puts up red flags in my head as being maybe _**possible**_ but not _**reasonable**_ to try and fix.

Reply:
I was probably thinking of more z-indexception 😬 .. background color styles go on a pseudo element with an even lower z-index than the shadows more so that adding padding based on shadow offset. But as Tyler pointed out, you're never forced into the issue as you can always manually add section padding to avoid it. So let's ignore that for now.